// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
package com.oppo.ohacker.player;

// The player service definition.
service PlayerService {
    rpc initChessBoard (InitRequest) returns (InitResponse) {
    }

    rpc move (MoveRequest) returns (MoveResponse) {
    }
}

// The request message
message InitRequest {
    bool isFirstHand = 1;
}

message Chess {
    int32 chessId = 1;
    int32 value = 2;
}

message Position {
    int32 x = 1;
    int32 y = 2;
}

message ChessPosition {
    Position position = 1;
    Chess chess = 2;
}

message MoveRequest {
    repeated ChessPosition chessPositions = 1;
    bool isFirstHand = 2;
}

// The response message
message InitResponse {
    repeated ChessPosition chessPosition = 1;
}

message MoveResponse {
    repeated Position position = 1;
}